minishift/minishift
# Issue #2601 Append kubeconfig setting instead merge

Fixes #2601 


Steps to test the pull request

  1. create a self cert or use 
```
$ cat cluster-ca.crt 
-----BEGIN CERTIFICATE-----
MIIC6jCCAdKgAwIBAgIBATANBgkqhkiG9w0BAQsFADAmMSQwIgYDVQQDDBtvcGVu
c2hpZnQtc2lnbmVyQDE1MzcxNjQwODQwHhcNMTgwOTE3MDYwMTIzWhcNMjMwOTE2
MDYwMTI0WjAmMSQwIgYDVQQDDBtvcGVuc2hpZnQtc2lnbmVyQDE1MzcxNjQwODQw
ggEiMA0GCSqGSIb3DQEBAQUAA4IBDwAwggEKAoIBAQCpHq5eG6MZJJZVIMJik7Hv
Rqr45OT7EZ1+U7H3Nv96bCSzZoVbwPUg56x8+323IqHwFJoFoeVso8Q9hwkvX50U
NfLPgjTAbfZKIhefgjjbA92sBiLPFNKiNrLGMrqqqIGTrUFsKT52QQptNuX/++XA
gAQTmQpmwOUv/Vi98AeaWArZEjMjF8DvosCuU4olnGgrIJ9z0yUuXNtFpi15hhVu
2f1Wtk0/YLvIu11ZyJdZACsbHELEQWWNVXgXzZVaQO5ql4Y2ApKJdgO953ADicuu
OuMo1eqoPBmBJvq++/ynE6LXRFxLkuIrdzV6ujepboQcPrYuihBaBWNI7dnEsWcj
AgMBAAGjIzAhMA4GA1UdDwEB/wQEAwICpDAPBgNVHRMBAf8EBTADAQH/MA0GCSqG
SIb3DQEBCwUAA4IBAQBKATFFOTZbD5nJ/zDaPgk+zAv5nZ8Fx8MijzQkMXYrrkiz
T5Vnx2jXuxFeWcpJ2Lj7JzwYIGdNQb25xG1bRwCFYpu8XE+SKVgIFAwE7wNyDR2S
RdiyY3FVT2oKiporD+aWxfaiWTSUhniaezyJcrPOUPlnQRqzwAO/bmDWY19xkQ9f
ynvDAcYg3CzyK3DkquOoeCG5CZa5rsAJVUAIFT4BXq9h2n7HG9e9e9DyVxgT7rB3
FdEJZ4+s0dG4+HEg1e4cIseE/SkemWGq7YEGBjjr2vsgQMqqlaXmbExNHcATeSEV
flPMVFHuJZwQD4i/qdDIoZ8D1cd2oBeXEcPfT0fD
-----END CERTIFICATE-----
```
  2. Delete the kube config file
```
$ rm -fr ~/.kube
```
  3.  Add a random cluster using this cert
```
$ oc config set-cluster 192-168-42-144:8443 --server https://192.168.42.144:8443 --certificate-authority cluster-ca.crt 
```
  4. check the config file
```
$ cat ~/.kube/config 
apiVersion: v1
clusters:
- cluster:
    certificate-authority: /home/prkumar/work/github/practice/go/bin/cluster-ca.crt
    server: https://192.168.42.144:8443
  name: 192-168-42-144:8443
contexts: []
current-context: ""
kind: Config
preferences: {}
users: []
```
  5. start minishift without using this patch. check the config again it subsitute the path of certificate-authority to actual cert for `192-168-42-144:8443` which we just added in step-4

  6. repeat till step 4 and use this patch and again check the config file it will not subsituted now.
